content/guides/developer-guides/delfin/_index.md

```diff
@@ -38,17 +42,190 @@ Existing Delfin Drivers for reference:
+| **remove_trap_config()** | `context`: Delfin framework context, `trap_config`: trap configuration | None |  Remove trap receiver configuration from storage system |
+| **parse_alert()** | `context`: Delfin framework context, `alert`: alert to parse | Alert dict as shown in `delfin/drivers/driver.py` |  Parse alert data got from snmp trap server |
+| **list_alerts()** | `context`: Delfin framework context, `query_para`: optional, contains 'begin_time' and 'end_time' | Alert dict as shown in `delfin/drivers/driver.py` |  List all current alerts from storage system |
+| **clear_alert()** | `context`: Delfin framework context, `alert`: alert to clear | Success/Failure |   Clear alert from storage system |
+
+  ```python
+
+    def get_storage(self, context):
+        # Do something here
+        return {
+            'name': 'sample',
+            'description': 'sample',
+            'vendor': 'sample_vendor',
+            'model': 'sample_model',
+            'status': 'normal',
+            'serial_number': 123,
+            'firmware_version': '1.0.0',
+            'location': 'HK',
+            'total_capacity': 123,
+            'used_capacity': 0,
+            'free_capacity': 123,
+            'raw_capacity': 123,
+            'subscribed_capacity': 123
+        }
+
+    def list_storage_pools(self, context):
+        pool_list = []
+        for idx in range(3):
+            p = {
+                "name": "sample_pool_" + str(idx),
+                "storage_id": self.storage_id,
+                "native_storage_pool_id": "sample_original_id_" + str(idx),
+                "description": "Sample Pool",
+                "status": "normal",
+                "total_capacity": 123,
+                "used_capacity": 0,
+                "free_capacity": 123,
+            }
+            pool_list.append(p)
+        return pool_list
+
+    def list_volumes(self, context):
+      volume_list = []
+      for i in range(1, 5):
+          v = {
+              "name": "sample_vol_" + str(i),
+              "storage_id": self.storage_id,
+              "description": "Sample Volume",
+              "status": "normal",
+              "native_volume_id": "sample_original_id_" + str(i),
+              "wwn": "fake_wwn_" + str(i),
+              "total_capacity": 12,
+              "used_capacity": 2,
+              "free_capacity": 10,
+          }
+          volume_list.append(v)
+          return volume_list
+
+    def add_trap_config(self, context, trap_config):
+        pass
+
+    def remove_trap_config(self, context, trap_config):
+        pass
+
+    def parse_alert(self, context, alert):
+        pass
+
+    def clear_alert(self, context, alert):
+        pass
+
+    def list_alerts(self, context, query_para=None):
+        pass
+
+  ```
+
+* Install delfin and start delfin services of api.py, task.py and alert.py.
+
+  ```bash
+  cd <delfin directory>
+  python setup.py install
+  python delfin/cmd/api.py --config-file ./etc/delfin/delfin.conf &
```

Comment:
Suggestion: a) nohup ?
b) Should it be suggested to put all these commands to run in BG inside a script and execute (Just to avoid any fg command issue and breaking it)

Reply:
Modified to use existing script in the repo